src/sentry/snuba/metrics/fields/base.py

```diff
@@ -221,10 +284,80 @@ def generate_available_operations(self):
     def generate_default_null_values(self):
         return DEFAULT_AGGREGATES[self.op]
 
-    def run_post_query_function(self, data, idx=None):
+    def run_post_query_function(self, data, query_definition: QueryDefinition, idx=None):
         key = f"{self.op}({self.metric_name})"
         return data[key][idx] if idx is not None else data[key]
 
+    def generate_bottom_up_derived_metrics_dependencies(self):
+        return [(self.op, self.metric_name)]
+
+
+class HistogramMetricField(RawAggregatedMetric):
+    def generate_orderby_clause(
+        self, projects: Sequence[Project], direction: Direction
+    ) -> List[OrderBy]:
+        raise DerivedMetricParseException("histograms cannot be ordered")
+
+    def generate_default_null_values(self):
+        return []
+
+    def generate_available_operations(self):
+        return [self.op]
+
+    def __resize_histogram(
+        self, data: ClickhouseHistogram, query_definition: QueryDefinition
+    ) -> ClickhouseHistogram:
+        if not data:
+            return data
+
+        # Get lower and upper bound of data. If the user defined custom ranges,
+        # honor them.
+        data.sort()
+        min_val = data[0][0]
+        max_val = data[-1][1]
+        if query_definition.histogram_from is not None:
+            min_val = max(min_val, query_definition.histogram_from)
+        if query_definition.histogram_to is not None:
+            max_val = min(max_val, query_definition.histogram_to)
+        bucket_count = query_definition.histogram_buckets
+
+        buckets = [
+            (
+                min_val + (max_val - min_val) * (x / bucket_count),
+                min_val + (max_val - min_val) * ((x + 1) / bucket_count),
+            )
+            for x in range(bucket_count)
+        ]
+
+        rv = {bucket: 0 for bucket in buckets}
+
+        # XXX: quadratic function
```

Comment:
this hideous function was copied from the histogram experiments repo. I believe it will perform fine for now because we're going to do `250*250=62k` iterations at max, which is probably still manageable in endpoints

Reply:
Should we create a span around this loop so we can measure how long it takes?